treeDetectLmax.py

```diff
@@ -0,0 +1,276 @@
+        # Create point
+        geometry = ogr.Geometry(ogr.wkbPoint)
+        geometry.SetPoint(0, geoLocation[0], geoLocation[1])
+        geometry.Transform(transform)
+        
+        # Create feature
+        feature = ogr.Feature(layer.GetLayerDefn())
+        feature.SetGeometry(geometry)
+        feature.SetFID(pointIndex)
+        feature.SetField("H",float(height[pointIndex]))
+        
+        # Save feature
+        layer.CreateFeature(feature)
+        # Cleanup
+        geometry.Destroy()
+        feature.Destroy()
+    # Cleanup
+    kmlData.Destroy()
+    # Return
+    return kmlPath
+    
+
+def getSpatialReferenceFromWkt(Wkt):
+    '''Return GDAL spatial reference object from proj4 string'''
+    spatialReference = osr.SpatialReference()
+    spatialReference.ImportFromWkt(Wkt)
+    return spatialReference
+    
+def validateShapePath(shapePath):
+    '''Validate shapefile extension'''
+    return os.path.splitext(str(shapePath))[0] + '.shp'
+    
+def validateKmlPath(kmlPath):
+    '''Validate kml extension'''
+    return os.path.splitext(str(kmlPath))[0] + '.kml'    
+    
+# convert pixel coordinates to geospatial coordinates
+def ApplyGeoTransform(inx,iny,gt):
+    ''' Apply a geotransform
+        @param  inx       Input x coordinate (double)
+        @param  iny       Input y coordinate (double)
+        @param  gt        Input geotransform (six doubles)
+        @return outx,outy Output coordinates (two doubles)
+    '''
+    outx = gt[0] + inx*gt[1] + iny*gt[2]
+    outy = gt[3] + inx*gt[4] + iny*gt[5]
+    return (outx,outy)    
+
+
+# if __name__ == "__main__":
+    
+    # options = {
+    # 'WinRad': float(sys.argv[2]), 
+    # 'MinHeightThres': float(sys.argv[3]),
+    # 'suffix': str(sys.argv[1]),
+    # 'src': str(sys.argv[4]),
+    # 'dst': str(sys.argv[5])
+    # }
+    # print options
+    # print "--main a été activée"
+    # if not options['src'].endswith('/'):
+        # options['src'] = options['src'] + '/' 
+        
+    # if not options['dst'].endswith('/'):
+        # options['dst'] = options['dst'] + '/'         
+        
+    # if os.path.isdir(options['src']) == False:
+        # main(options)
+    # if os.path.isdir(options['src']) == True:
+        # file_list = os.listdir(options['src'])
+        # inputDir = options['src']
+        # shpdst = options['dst'] + 'shp'
+        # kmldst = options['dst'] + 'kml'
+        # if not os.path.exists(shpdst):
+            # os.makedirs(shpdst)
+            # print 'output folder ' + shpdst + ' was created'
+        # if not os.path.exists(kmldst):
+            # os.makedirs(kmldst)
+            # print 'output folder ' + kmldst + ' was created'
+        # for k, file_list in enumerate(file_list):
+            # print('processing ' + file_list)
+            # options['src'] = inputDir + file_list
+            # main(options)
+
+__author__ = "Matthew Parkan, LASIG, EPFL"
+__license__ = "GPL"
+__version__ = "1.0.1"
```

Comment:
It is actually a modified version since it is adapted to run from a Qgis plugin call